joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstForFunctionsCreator.scala

```diff
@@ -268,8 +290,20 @@ trait AstForFunctionsCreator(implicit withSchemaValidation: ValidationMode) { th
 
         val stmtBlockAst = astForMethodBody(node.body, optionalStatementList)
 
+        val anonProcParam = scope.anonProcParam.map { param =>
+          val paramNode = ProcParameter(param)(node.span.spanStart(s"&$param"))
+          val index =
+            parameterAsts.lastOption.flatMap(_.root).map { case m: NewMethodParameterIn => m.index }.getOrElse(1)
```

Comment:
This `getOrElse` is at `1`, while the above is at `0`, is this intentional? I would think, on the condition that you should append `1` to index at `astForParameter`, that the truth should be `0`

Reply:
I should add that to the test.

Reply:
But the off by one error is not.